daos/GroupDao.ts

```diff
@@ -0,0 +1,94 @@
+/**
+ * @file Implements DAO managing data storage of groups. Uses mongoose GroupModel
+ * to integrate with MongoDB
+ */
+import GroupDaoI from "../interfaces/group/GroupDaoI";
+import Group from "../models/group/Group";
+import GroupModel from "../mongoose/groups/GroupModel"
+
+/**
+ * @class GroupDao Implements Data Access Object managing data storage
+ * of Groups
+ * @property {GroupDao} groupDao Private single instance of GroupDao
+ */
+export default class GroupDao implements GroupDaoI {
+    private static groupDao: GroupDao | null = null
+
+    /**
+     * Creates singleton DAO instance
+     * @returns {GroupDao} GroupDao
+     */
+    public static getInstance = (): GroupDao => {
+        if (GroupDao.groupDao === null) {
+            GroupDao.groupDao = new GroupDao()
+        }
+        return GroupDao.groupDao
+    }
+
+    private constructor() {
+    }
+
+    /**
+     * Removes group object with given primary key from database.
+     * @param {string} uid Primary key of user
+     * @param {string} gid Primary key of group
+     * @returns Promise To be notified when group is removed from the database
+     */
+    deleteGroup = async (uid: string, gid: string): Promise<any> =>
```

Comment:
In rethinking this - do we need a user id here?
Couldn't we just take the session's active user id to accomplish the same?

Reply:
Yes we can probably just use gid to delete group and verify user in session is admin